commands/files/multipartfile.go

```diff
@@ -55,6 +55,22 @@ func NewFileFromPart(part *multipart.Part) (File, error) {
 	return f, nil
 }
 
+func NewFileFromRequest(r *http.Request) (File, error) {
+	contentType := r.Header.Get(contentTypeHeader)
+	mediaType, _, _ := mime.ParseMediaType(contentType)
```

Comment:
err ignored due to https://github.com/ipfs/go-ipfs/commit/ce49541f130e736639ff270ad0132245f2938ee8. If there is a more semantic way to express that this is run only when files are involved.